skfp/fingerprints/map.py

```diff
@@ -139,135 +144,124 @@ def __init__(
         )
         self.fp_size = fp_size
         self.radius = radius
-        self.variant = variant
+        self.include_duplicated_shingles = include_duplicated_shingles
+        self.counts = counts
 
     def _calculate_fingerprint(
         self, X: Sequence[Union[str, Mol]]
     ) -> Union[np.ndarray, csr_array]:
         X = ensure_mols(X)
         X = np.stack(
-            [self._calculate_single_mol_fingerprint(mol) for mol in X], dtype=int
+            [self._calculate_single_mol_fingerprint(mol) for mol in X],
+            dtype=np.uint32 if self.counts else np.uint8,
         )
 
-        if self.variant == "bit":
-            X = (X > 0).astype(np.uint8)
-        elif self.variant == "count":
-            X = X.astype(np.uint32)
-
         return csr_array(X) if self.sparse else np.array(X)
 
     def _calculate_single_mol_fingerprint(self, mol: Mol) -> np.ndarray:
         atoms_envs = self._get_atom_envs(mol)
-        shingles = self._get_atom_pair_shingles(mol, atoms_envs)
+        shinglings = self._get_atom_pair_shingles(mol, atoms_envs)
+
+        if self.counts:
+            folded = np.zeros(self.fp_size, dtype=np.uint32)
+            for shingling in shinglings:
+                hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
+                folded[hashed % self.fp_size] += 1
+            return folded
+
+        folded = np.zeros(self.fp_size, dtype=np.uint8)
+        for shingling in shinglings:
+            hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
+            folded[hashed % self.fp_size] = 1
+        return folded
+
+    @classmethod
+    def _find_env(cls, mol: Mol, atom_identifier: int, radius: int) -> Optional[str]:
+        """Returns a smile representation of the atom environment of a given radius.
+
+        Parameters
+        ----------
+        mol : Mol
+            The molecule to calculate the fingerprint for
+        atom_identifier : int
+            The index of the atom to calculate the environment for
+        radius : int
+            The radius of the environment
+
+        Returns
+        -------
+        str
+            The SMILES representation of the atom environment, or
+            None if the atom is not found
```

Comment:
We generally don't put full docstrings in private methods, since they would have to be maintained, and private methods in general could be changed without impacting the public API. Just the first sentence of this would suffice.

Reply:
Kept only first line.